src/gui/render/array_buffer/array_buffer.hpp

```diff
@@ -31,35 +34,48 @@ enum class GPUDataType : GLenum {
     // clang-format on
 };
 
-enum class buffer_type : GLenum {
+/**
+ * @brief Targe. Designates how the buffer is used.
+ */
+enum class BindingTarget : GLenum {
     ARRAY_BUFFER = GL_ARRAY_BUFFER,
     ELEMENT_ARRAY_BUFFER = GL_ELEMENT_ARRAY_BUFFER,
 };
 
-class ArrayBuffer;
-
-// cursed
-class array_buffer_type {
- protected:
-    const uint8_t vec_size_;  // 1,2,3 or 4 vec size
-    const uint8_t type_size_; // size of type float, int etc.
-    const bool is_int_;       // should be interpreted as integer
-    const GPUDataType draw_type_;
-
- public:
-    inline constexpr array_buffer_type(
+/**
+ * @brief Determines how a data type should be interpreted on the GPU.
+ *
+ * @details The entire thing works by template deduction and magic.
+ */
+struct GPUArrayType {
```

Comment:
this is not POD

```suggestion
class GPUArrayType {
```

Reply:
Everything is either static or an initializer. I had it like this because I was using it as a template parameter.

Reply:
> Everything is either static or an initializer.

Thus, not POD.



> I had it like this because I was using it as a template parameter.

Shouldn't matter